Sync/StorageClient.swift

```diff
@@ -605,6 +616,210 @@ public class Sync15StorageClient {
     }
 }
 
+public class Sync15BatchClient<T: CleartextPayloadJSON> {
+    private let config: InfoConfiguration
+
+    private var records: [Record<T>] = []
+    private let collectionClient: Sync15CollectionClient<T>
+    private let collectionURI: NSURL
+
+    init(config: InfoConfiguration, collectionURI: NSURL, collectionClient: Sync15CollectionClient<T>) {
+        self.config = config
+        self.collectionURI = collectionURI
+        self.collectionClient = collectionClient
+    }
+
+    public func addRecords(records: [Record<T>]) {
+        log.debug("Adding \(records.count) records into batch")
+        self.records += records
```

Comment:
If the server supports batching, you should consider chaining out batches as they fill, right here. It'll need a little dancing around a `Deferred`, of course. Follow-up is fine.